python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/router_bgp.py

```diff
@@ -36,40 +35,26 @@ def router_bgp(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol) -> dict
                 msg = f"{self.data_model}.p2p_links.[].as or {self.data_model}.p2p_links_profiles.[].as"
                 raise AristaAvdInvalidInputsError(msg)
 
-            neighbor = {
-                "remote_as": p2p_link_data["peer_bgp_as"],
-                "peer": p2p_link_data["peer"],
-                "description": p2p_link_data["peer"],
-                "peer_group": self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
-            }
+            neighbor = EosCliConfigGen.RouterBgp.NeighborsItem(
+                remote_as=p2p_link_data["peer_bgp_as"],
+                peer=p2p_link_data["peer"],
+                description=p2p_link_data["peer"],
+                peer_group=self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
+            )
 
             # RFC5549
             if self.inputs.underlay_rfc5549 and p2p_link.routing_protocol != "ebgp":
-                neighbor_interfaces.append({"name": p2p_link_data["interface"], **neighbor})
+                self.structured_config.router_bgp.neighbor_interfaces.append_new(name=p2p_link_data["interface"], **neighbor._as_dict())
```

Comment:
We should avoid going to dict at all.
It will be simpler to just build the full neighbor interface here setting the 5 arguments directly.
And then remove the `neighbor = ` stuff below this `if` so it only runs when building a regular neighbor.

Alternative is to use _cast_as

Reply:
Fixed!